### DISCUSSION TITLE
Approach to exporting / importing styles

### DISCUSSION THREAD

**Discussion by dpeek** [johanholmerin/style9#33] [Q&A] [answered]
I thought I'd open this as a discussion as I could either be missing something or it could be a limitation of the compile time approach.

Is there a way to use an imported style9 object in another file? I was testing out the library by building out a small UI Kit and ran into this error:

```
Return value from style9.create has to be called as a function or accessed as an object
```

I think I understand the limitation, just wondering how best to work around it for some centralised styles (colours, spacing, etc). I can see how you could export the individual class names for use, perhaps there could be a way to export the top level named classes as a simple object?

**Comment by johanholmerin** [comment, marked as answer]
I assume you're trying to do something like this:

```javascript
export const styles = style9.create({
  blue: {
    color: 'blue',
  },
  red: {
    color: 'red',
  }
});
```
The reason this doesn't work is because `styles` can be used as both as a function or an object, e.g. `styles('blue')` or `styles.blue`. However, the the function use case only exists in the compiler and the function calls get replaced with the resolved class names. This can't be done across files, but Typescript would still recognise it as a function when imported, which is why there are limitations on how you can use the variable. If you want access to the object, e.g. for shared styles, you can use spread:

```javascript
export const { ...styles } = style9.create({
  // ...
});
```

You could also access one specific property:

```javascript
export const { blue } = style9.create({
  // ...
});
export const red = style9.create({
  // ...
}).red;
```

**Reply by dpeek** [reply]
Thanks for the quick reply :) Unfortunately the object spread approach doesn't seem to work, giving me this error:

```
SyntaxError: .../theme.ts: Return value from style9.create has to be called as a function or accessed as an object
  26 |   }
  27 | }),
> 28 |     theme = _extends({}, _style9$create);
```

Export:

```typescript
export const { ...theme } = style9.create({
  control: { ... }
}
```

Import:

```typescript
import { theme } from './theme';
console.log(theme.control);
```

Specific exports do work though and will do fine for now :)

Thanks for the great library :)

**Reply by johanholmerin** [reply]
Which bundler are you using, webpack? The issue is that you run style9 after Babel transform, hence the `theme = _extends({}, _style9$create);`. Could you create a repo with your setup?

**Reply by dpeek** [reply]
Conveniently I've been working directly in the nextjs example folder of this repo :) Can commit a repro to my fork if that helps?

**Reply by johanholmerin** [reply]
Fixed in 0.9.0
